test/suite/test_import01.py

```diff
@@ -74,16 +74,18 @@ def check(self, uri, keys, values):
             else:
                 self.check_record(uri, keys[i], values[i])
 
-    # We know the ID can be different between configs, so just remove it from comparison.
-    # Everything else should be the same.
+    # The ID and checkpoint information can be different between configs, remove it. Everything else
+    # should be the same.
     def config_compare(self, aconf, bconf):
         a = re.sub('id=\d+,?', '', aconf)
-        a = (re.sub('\w+=\(.*?\)+,?', '', a).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', a))
+        a = re.sub('checkpoint=\(.*?\)+,?', '', a)
         b = re.sub('id=\d+,?', '', bconf)
-        b = (re.sub('\w+=\(.*?\)+,?', '', b).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', b))
-        self.assertTrue(a.sort() == b.sort())
+        b = re.sub('checkpoint=\(.*?\)+,?', '', b)
```

Comment:
If this is a bug, it would have existed before this change too. But wouldn't this match too far? i.e. given a string `checkpoint=(some,stuff),shouldnt_be_deleted=true,b=(some,other,stuff)` - it would match the entire string rather than just the `checkpoint=(some,stuff)` clause?

Reply:
Hmm, it seems to work in a REPL. I don't know why it's not being "greedy" like I'd expect.

Reply:
Oh, it's the `.*?` construct - I get it now.

Reply:
I ran `test_import01` and dumped `aconf`. The checkpoint field looks like
```
checkpoint=( 
    WiredTigerCheckpoint.1=( 
        addr="018181e40de2a8838281e488c2b6928383e48aff1d60808080cdc0c1c0",
        order=1,
        time=1700608072,
        size=1024,
        newest_start_durable_ts=60,
        oldest_start_ts=10,
        newest_txn=8,
        newest_stop_durable_ts=0,
        newest_stop_ts=-1,
        newest_stop_txn=-11,
        prepare=0,
        write_gen=3,
        run_write_gen=1
    )
),
```
which means the greedy parsing will miss the final brace. We'll also run into trouble if the checkpoint config contains multiple checkpoint (i.e. named checkpoints). I'll think on this but I'm not sure regex is enough to handle the edge cases

Reply:
There was definitely an issue with the regex and having multiple (named) checkpoints. I reverted the regex and now explicitly remove checkpoints not using the regex as you suggested in 0e8742b3083361e195bd313aeafeccf5cf102745.